pythia/cleaning/midnight_rotation.py

```diff
@@ -0,0 +1,111 @@
+import datetime
+
+import astropy.units as u
+from sunpy.physics.differential_rotation import diff_rot
+
+__all__ = ['MidnightRotation']
+
+
+class MidnightRotation:
```

Comment:
I believe this would be a lot more useful if instead to Midnight it's to a user specified time. The user case for other people may be that they may have things that they want to rotate to a particular time, instead to midnight.

Reply:
:thinking: it may be a bit more difficult. Make an issue and we can revisit it on the future.

Reply:
But that works with the differential rotation anyway. This module does the additional computation of getting the nearest midnight. If it's just a user-specified time, then the user can directly use the differential rotation.

Reply:
I'm sure there are use cases where I'd need to move all to noon instead to midnight (I imagine people working with ground observatories may want that). But as I said, we can create an issue and revisit later if needed.

Reply:
#39